src/main/scala/io/iohk/ethereum/vm/ProgramState.scala

```diff
@@ -0,0 +1,58 @@
+package io.iohk.ethereum.vm
+
+import akka.util.ByteString
+
+object ProgramState {
+  def apply(context: ProgramContext): ProgramState =
+    ProgramState(context = context, storage = context.storage)
+}
+
+/**
+  * Intermediate state updated with execution of each opcode in the program
+  *
+  * @param context the context which initiates the program
+  * @param stack current stack
+  * @param memory current memory
+  * @param storage current storage
+  * @param pc program counter - an index of the opcode in the program to be executed
+  * @param returnData data to be returned from the program execution
+  * @param halted a flag to indicate program termination
+  * @param error indicates whether the program terminated abnormally
+  */
+case class ProgramState(
+  context: ProgramContext,
+  stack: Stack = Stack.empty(),
+  memory: Memory = Memory.empty,
+  storage: Storage,
```

Comment:
I guess context.storage is the initial context supplied to VM, while this field changes throughout code execution, am I right?